packages/core/src/model.types.ts

```diff
@@ -37,10 +37,21 @@ export interface Model<
   events: Prop<TModelCreators, 'events'>;
   actions: Prop<TModelCreators, 'actions'>;
   reset: () => AssignAction<TContext, any>;
-  createMachine: (
-    config: MachineConfig<TContext, any, TEvent, TAction>,
-    implementations?: Partial<MachineOptions<TContext, TEvent, TAction>>
-  ) => StateMachine<TContext, any, TEvent>;
+  createMachine: {
+    <TMeta extends DefaultTypegenMeta>(
+      config: MachineConfig<TContext, any, TEvent, TAction, TMeta>,
+      implementations?: Partial<
+        MaybeTypegenMachineOptions<TContext, TEvent, TMeta, TAction>
```

Comment:
`TMeta` and `TAction` switched here is a bit confusing (nit) - whereas it's `TAction` and `TMeta` above